mods/boats/init.lua

```diff
@@ -180,15 +171,13 @@ function boat.on_step(self, dtime)
 		self.object:set_pos(self.object:get_pos())
 		return
 	end
-	local s = get_sign(self.v)
-	self.v = self.v - dtime * 0.6 * s
-	if s ~= get_sign(self.v) then
-		self.object:set_velocity({x = 0, y = 0, z = 0})
+	-- We need to preserve velocity sign to properly apply drag force while moving backward
+	local drag = dtime * math.sign(self.v) * (0.01 + 0.08 * self.v * self.v)
+	-- If drag is larger than velocity, then stop horizontal movement.
+	if math.abs(self.v) <= math.abs(drag) then
 		self.v = 0
-		return
-	end
-	if math.abs(self.v) > 5 then
```

Comment:
There is no need to artificially limit the maximum speed of the boat. It is limited by the drag force.

Reply:
Correct =)